overlord/snapstate/snapstate.go

```diff
@@ -2995,49 +2675,52 @@ func autoRefreshPhase1(ctx context.Context, st *state.State, forGatingSnap strin
 }
 
 // autoRefreshPhase2 creates tasks for refreshing snaps from updates.
-func autoRefreshPhase2(ctx context.Context, st *state.State, updates []*refreshCandidate, flags *Flags, fromChange string) (*UpdateTaskSets, error) {
```

Comment:
I think some of these unused contexts were intentional because we wanted to propagate them eventually? May be misremembering though

Reply:
I think in this case the only thing that could reasonably be canceled is `checkDiskSpace`, which might hit the store (but usually won't). I'll add a TODO on that function signature? Either way though, all call sites of `autoRefreshPhase2` were creating their own contexts, so it having one as a parameter isn't really useful right now.